pkg/adapter/apiserver/adapter_test.go

```diff
@@ -28,82 +28,84 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/dynamic"
 	dynamicfake "k8s.io/client-go/dynamic/fake"
+	"knative.dev/eventing/pkg/adapter"
 	kncetesting "knative.dev/eventing/pkg/kncloudevents/testing"
 	rectesting "knative.dev/eventing/pkg/reconciler/testing"
+	pkkrectesting "knative.dev/pkg/reconciler/testing"
 	"knative.dev/pkg/source"
 )
 
 type mockReporter struct {
 	eventCount int
 }
 
+var (
+	fakeMasterURL = "test-source"
+)
+
 func (r *mockReporter) ReportEventCount(args *source.ReportArgs, responseCode int) error {
 	r.eventCount += 1
 	return nil
 }
 
 func TestNewAdaptor(t *testing.T) {
 	ce := kncetesting.NewTestClient()
-	logger := zap.NewExample().Sugar()
-	k8s := makeDynamicClient(nil)
+
+	masterURL = &fakeMasterURL
 
 	testCases := map[string]struct {
-		opt    Options
+		opt    envConfig
 		source string
 
 		wantMode      string
 		wantNamespace string
 		wantGVRCs     []GVRC
 	}{
-		"empty opts": {
-			opt:      Options{},
-			wantMode: RefMode,
-		},
 		"with source": {
 			source:   "test-source",
-			opt:      Options{},
+			opt:      envConfig{},
 			wantMode: RefMode,
 		},
 		"with namespace": {
 			source: "test-source",
-			opt: Options{
-				Namespace: "test-ns",
+			opt: envConfig{
+				EnvConfig: adapter.EnvConfig{
+					Namespace: "test-ns",
+				},
 			},
 			wantMode:      RefMode,
 			wantNamespace: "test-ns",
 		},
 		"with mode resource": {
 			source: "test-source",
-			opt: Options{
+			opt: envConfig{
 				Mode: ResourceMode,
 			},
 			wantMode: ResourceMode,
 		},
 		"with mode ref": {
 			source: "test-source",
-			opt: Options{
+			opt: envConfig{
 				Mode: RefMode,
 			},
 			wantMode: RefMode,
 		},
 		"with mode trash": {
 			source: "test-source",
-			opt: Options{
+			opt: envConfig{
 				Mode: "trash",
 			},
 			wantMode: RefMode,
 		},
 		"with mode gvrs": {
 			source: "test-source",
-			opt: Options{
-				GVRCs: []GVRC{{
-					GVR: schema.GroupVersionResource{
-						Group:    "apps",
-						Version:  "v1",
-						Resource: "replicasets",
-					},
-					Controller: true,
-				}},
+			opt: envConfig{
+				ApiVersion:      StringList{"apps/v1"},
```

Comment:
not related to this PR, but I was wondering if we have UTs for lists with more than 1 value?

Reply:
added one.